packages/fuel-indexer-schema/src/join.rs

```diff
@@ -0,0 +1,110 @@
+use crate::FtColumn;
+use fuel_indexer_lib::join_table_typedefs_name;
+use serde::{Deserialize, Serialize};
+
+extern crate alloc;
+
+/// Details for the many-to-many relationship.
+///
+/// This is essentially the same as `fuel_indexer_lib::graphql::parser::JoinTableRelation`, just
+/// a bit more compile-time friendly.
+#[derive(Debug, Clone)]
+pub struct JoinMetadata<'a> {
+    /// Name of join table.
+    pub table_name: &'a str,
+
+    /// Fully qualified GraphQL namespace for indexer.
+    pub namespace: &'a str,
+
+    /// Name of parent table's column on which join is applied.
+    pub parent_column_name: &'a str,
+
+    /// Name of child table's column on which join is applied.
+    pub child_column_name: &'a str,
+
+    /// Position in the parent's set of `FtColumn`s, at which the many-to-many child column is found.
+    pub child_position: usize,
+}
+
+/// A raw SQL query.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct RawQuery(pub String);
+
+impl From<RawQuery> for Vec<u8> {
+    fn from(query: RawQuery) -> Self {
+        query.0.into_bytes()
+    }
+}
+
+impl RawQuery {
+    pub fn query(&self) -> &str {
+        &self.0
+    }
+
+    /// Whether or not there are actual records to insert for this query.
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    /// Create a new `RawQuery` from the given metadata and columns.
+    pub fn from_metadata(metadata: &JoinMetadata<'_>, columns: &[FtColumn]) -> Self {
```

Comment:
I think we should make it clear that this is for `JoinMetadata` or rename `RawQuery` to something that makes it known that this is pretty much only for the parent-child join table stuff (as far as I can tell).